7.js

```diff
@@ -0,0 +1,61 @@
+const months = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun',
+'Jul', 'Aug', 'Sep', 'Oct', 'Nov',
+'Dec']
+
+function formatDate (date) {
+  if (date.includes('-')) {
```

Comment:
There's *a lot* of logic happening in this `if` block.  I suspect there is some opportunity for refactoring, especially since the `return` pattern looks fairly consistent...